benchmark-graphs/src/main/java/org/conscrypt/graphgen/Main.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.conscrypt.graphgen;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+import com.bazaarvoice.jolt.Chainr;
+import com.bazaarvoice.jolt.JsonUtils;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Utility to convert from the JMH JSON output to an HTML file.
+ */
+public class Main {
+
+  public static final String JSON_TEMPLATES = "/json/templates/";
+  public static final String HTML_TEMPLATES = "/html/";
+
+  public static void main(String[] args) throws IOException, URISyntaxException {
+    if (args.length != 3) {
+      System.err.println("Usage: graphgen [template] [input.json] [output.html]");
+      listAllResources(System.err);
+      System.exit(1);
+    }
+
+    try (InputStream spec = Main.class.getResourceAsStream(JSON_TEMPLATES + args[0]);
+        InputStream jmhIn = new BufferedInputStream(new FileInputStream(args[1]));
+        OutputStream output = new BufferedOutputStream(new FileOutputStream(args[2]))) {
+      writeHtml(output, "header.html");
+      convertJmhJsonData(spec, jmhIn, output);
+      writeHtml(output, "footer.html");
+    }
```

Comment:
Sorry, I'm not seeing it (probably blind) ... where do we close `output` to spark the `flush`?